File: app/lib/backend/schema/folder.dart
class Folder {
  final String id;
  final String uid;
  final String name;
  final String? color;
  final String? icon;
  final int position;
  final DateTime createdAt;
  final DateTime updatedAt;

  Folder({
    required this.id,
    required this.uid,
    required this.name,
    this.color,
    this.icon,
    required this.position,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Folder.fromJson(Map<String, dynamic> json) {
    return Folder(
      id: json['id'],
      uid: json['uid'],
      name: json['name'],
      color: json['color'],
      icon: json['icon'],
      position: json['position'] ?? 0,
      createdAt: DateTime.parse(json['created_at']).toLocal(),
      updatedAt: DateTime.parse(json['updated_at']).toLocal(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'uid': uid,
      'name': name,
      'color': color,
      'icon': icon,
      'position': position,
      'created_at': createdAt.toUtc().toIso8601String(),
      'updated_at': updatedAt.toUtc().toIso8601String(),
    };
  }
}

File: backend/models/folder.py
from datetime import datetime, timezone
from typing import Optional, List
from pydantic import BaseModel, Field


class Folder(BaseModel):
    id: str = Field(description="Unique identifier for the folder")
    uid: str = Field(description="User ID who owns this folder")
    name: str = Field(description="Name of the folder")
    color: Optional[str] = Field(default=None, description="Hex color code for the folder (e.g., #FF5733)")
    icon: Optional[str] = Field(default=None, description="Icon name or emoji for the folder")
    position: int = Field(default=0, description="Display position/order of the folder")
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))


class CreateFolder(BaseModel):
    name: str = Field(description="Name of the folder")
    color: Optional[str] = Field(default=None, description="Hex color code for the folder")
    icon: Optional[str] = Field(default=None, description="Icon name or emoji for the folder")
    position: int = Field(default=0, description="Display position/order of the folder")


class UpdateFolder(BaseModel):
    name: Optional[str] = None
    color: Optional[str] = None
    icon: Optional[str] = None
    position: Optional[int] = None


class ReorderFolders(BaseModel):
    folder_ids: List[str] = Field(description="Ordered list of folder IDs")
